pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -1422,12 +1422,17 @@ func constructDatastoreContext(config types.DownloaderConfig, status *types.Down
 }
 
 func rktFetch(url string, localConfigDir string) (string, error) {
+	urlSlice := strings.SplitAfterN(url, "/", 2)
+	imageName := urlSlice[1]
```

Comment:
This can panic if the url argument is not formed as you expect. Need at least a check for len(urlSlice).

In general it is much better to have the caller pass in what you need instead of having to take apart the URL; we've had bugs due to this in the downloader already. 
How do you know how many components there are in the URL in this case?

Reply:
We are preparing the download URL from the datastore context and image context. We are taking FQDN from the datastore config and imageName from the image config.
I will add check for len(urlSlice).

Reply:
Why not pass in the arguments to the function instead of constructing an URL and then having the function have to (possibly incorrect!) take apart that URL?
We recently had to fix this for other URLs in the downloader. And I fear you are introducing the same type of bug in this code.